### PR TITLE
Add an icon in the list view for published notes

### LINKED ISSUE #220: Add icon to note list item if it is published
Similar to the web app.

### PR DESCRIPTION
**Pull request by dmsnell**
Resolves #220

Previously this app lacked the icon that the existing Simplenote web app
has for notes that are published.

This path adds the "signal" or "feed" icon to the left of the note
snippet in the list view for those notes.

**Before**
screen shot 2016-03-15 at 10 28 35 am

**After**
screen shot 2016-03-15 at 10 28 50 am

cc: @drw158 for CSS review. I think I didn't break anything :smile:

### CODE REVIEW

File: scss/note-list.scss [hunk 1]
```diff
@@ -152,6 +153,18 @@
 		}
 	}
 
+	.note-list-item-published-icon {
+		position: absolute;
+		left: 13px;
```
Comment (drw158) [inline scss/note-list.scss]: `left: 9px;` will make it look a bit more centered

File: scss/note-list.scss [hunk 2]
```diff
@@ -86,6 +86,7 @@
 }
 
 .note-list-item {
+	position: relative;
```
Comment (drw158) [inline scss/note-list.scss]: We could probably do some fancy flexbox here, but this works fine. No reason to make it complicated.

**Comment by drw158** [comment]
Dark theme looks good, everything works for me!

The RSS icon was a weird choice for "published". I can update it later, maybe before we launch v1.

**Comment by drw158** [comment]
Whoops, I forgot to test the Condensed setting. 

screen shot 2016-03-15 at 2 42 43 pm

**Comment by drw158** [comment]
I think something like this would be the best option for all note display settings

screen shot 2016-03-15 at 2 46 41 pm

We will need to take care of truncated text though.

screen shot 2016-03-15 at 2 47 52 pm

**Comment by drw158** [comment]
I got this working in the inspector

screen shot 2016-03-15 at 2 56 19 pm

**Comment by dmsnell** [comment]
thanks @drw158! and good catch on the condensed view. Do you mind sharing how you were able to get it to look right in the inspector?

**Comment by drw158** [comment]
Sure, I 

-inserted the svg into the title div `note-list-item-title`.
- wrapped the title text in a `span`.
- `display: flex` on `note-list-item-title` div.
- `margin-left: auto` on the svg.

The ellipsis doesn't work though at this point. Add this to the new span element with the title text:

- `width: calc(100% - 24px);`

That gets the truncation to work, although I'm not sure why. 24px is about the amount of space the icon takes up.

**Comment by dmsnell** [comment]
@drw158 I still had overlap even after playing with the `ellipsis` property. It's looking good though.

screen shot 2016-03-16 at 11 12 57 am

**Comment by drw158** [comment]
Hm! Maybe I left out a step.

**Comment by dmsnell** [comment]
@drw158 updated and the icon has been removed from the compact view. You will need to rebase if you want to test and probably wipe your node modules and reinstall with `npm install`

cc: @roundhill

**Comment by drw158** [comment]
Oh, hm I forgot that we had the icon positioned on the right to account for the condensed view. Looks like I had it working in the inspector at one point:

https://github.com/Automattic/simplenote-electron/pull/228#issuecomment-196996338

**Comment by dmsnell** [comment]
> I had it working in the inspector at one point:

@drw158 I think I had visual glitches pictured in the reply to that comment

**Comment by drw158** [comment]
Yea, I'm not sure how I got it working. I'd have to open it up and try it.

**Comment by roundhill** [comment]
I just noticed that in the web app we show the icon if a note is both `published` and `shared` with another user. Should we mirror that in this app?

**Comment by drw158** [comment]
We should eventually

**Comment by dmsnell** [comment]
@roundhill and @drw158 should we push this PR out the door? wait for matching the icon behavior for `published` _and_ `shared`?

**Comment by drw158** [comment]
Hm, I can't seem to get this branch working. I can't pull the latest because of merge conflicts. git reset --hard doesn't seem to work.

**Comment by drw158** [comment]
Got it working. I tweaked the CSS a bit; it wasn't working correctly.

screen shot 2016-10-07 at 11 43 22 am

Can we enable the icon for the condensed view? I think it is being hidden.

**Comment by dmsnell** [comment]
Bummer. We appear to have an issue persisting the public URL when sharing a note…

**Testing**
In **master** or in this branch, shouldn't matter.

Publish a note.
Watch the public URL appear.
Reload the app.
Note that the public URL is gone and won't come back.
Cry.

**Comment by roundhill** [comment]
> Cry.

😢 I wonder if was something we added recently, this wasn't reported earlier...

**Comment by roundhill** [comment]
Besides discovering #466, this looks good :shipit:

**Pull request merged**